runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -939,6 +980,7 @@ MM_CopyForwardScheme::reserveMemoryForObject(MM_EnvironmentVLHGC *env, UDATA com
 	if (NULL != result) {
 		regionList->_cacheAcquireCount += 1;
 		regionList->_cacheAcquireBytes += allocDescription.getBytesRequested();
+		setCompressedSurvivorCards(env, (void*)result, (void*) ((UDATA)result + allocDescription.getBytesRequested()));
```

Comment:
in some cases this could be a very large object (up to region size), but we probably need to set only 1 card, for the very beginning of object 
try providing just 'result' ( or 'result + min object size') as an end address